### PR TITLE
Add filter for Patient List

### PR DESCRIPTION
**Pull request by reshma-t**
Add filter on _disease_status_ for Patient List API
[Issue#258](https://github.com/coronasafe/care/issues/258)

### CODE REVIEW

File: care/facility/api/viewsets/patient.py [hunk 1]
```diff
@@ -51,16 +51,25 @@ def filter_list_queryset(self, request, queryset, view):
...
     serializer_class = PatientDetailSerializer
     filter_backends = (
         PatientDRYFilter,
         filters.DjangoFilterBackend,
     )
     filterset_class = PatientFilterSet
 
+    def get_queryset(self):
+        filter_query = self.request.GET.get('disease_status', None)
+        if (filter_query):
```
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:63]: Codacy Issue found: [Unnecessary "else" after "return"](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

File: care/facility/api/viewsets/patient.py [hunk 2]
```diff
@@ -51,16 +51,25 @@ def filter_list_queryset(self, request, queryset, view):
...
+            return PatientRegistration.objects.filter(disease_status=DiseaseStatusEnum[filter_query].value
+                                                      ).select_related(
+                "local_body", "district", "state", "facility", "facility__local_body", "facility__district", "facility__state"
```
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:66]: Codacy Issue found: [Wrong hanging indentation (add 42 spaces).](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

File: care/facility/api/viewsets/patient.py [hunk 3]
```diff
@@ -51,16 +51,25 @@ def filter_list_queryset(self, request, queryset, view):
...
+            )
+        else:
+            return PatientRegistration.objects.all().select_related(
+                "local_body", "district", "state", "facility", "facility__local_body", "facility__district", "facility__state"
```
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:70]: Codacy Issue found: [Line too long (126/120)](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

File: care/facility/api/viewsets/patient.py [hunk 4]
```diff
@@ -51,16 +51,25 @@ def filter_list_queryset(self, request, queryset, view):
...
     )
     filterset_class = PatientFilterSet
 
+    def get_queryset(self):
+        filter_query = self.request.GET.get('disease_status', None)
+        if (filter_query):
+            return PatientRegistration.objects.filter(disease_status=DiseaseStatusEnum[filter_query].value
+                                                      ).select_related(
+                "local_body", "district", "state", "facility", "facility__local_body", "facility__district", "facility__state"
+            )
```
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:67]: Codacy Issue found: [Wrong hanging indentation.](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

File: care/facility/api/viewsets/patient.py (hunk 1 above)
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:63]: Codacy Issue found: [Unnecessary parens after 'if' keyword](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

Comment (reshma-t) [reply care/facility/api/viewsets/patient.py:63]: Correcting

**Comment by vigneshhari** [comment]
The code breaks all security validations in place right ?, correct me if i am wrong.

File: care/facility/api/viewsets/patient.py [hunk 5]
```diff
@@ -51,16 +51,25 @@ def filter_list_queryset(self, request, queryset, view):
...
     )
     filterset_class = PatientFilterSet
 
+    def get_queryset(self):
+        filter_query = self.request.GET.get('disease_status', None)
+        if (filter_query):
+            return PatientRegistration.objects.filter(
+                disease_status=DiseaseStatusEnum[filter_query].value).select_related(
+                "local_body", "district", "state", "facility",
+                "facility__local_body", "facility__district", "facility__state")
```
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:67]: Codacy Issue found: [Wrong hanging indentation (add 4 spaces).](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

File: care/facility/api/viewsets/patient.py (hunk 1 above)
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:63]: Codacy Issue found: [Unnecessary parens after 'if' keyword](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

File: care/facility/api/viewsets/patient.py [hunk 6]
```diff
@@ -51,16 +51,25 @@ def filter_list_queryset(self, request, queryset, view):
...
         filters.DjangoFilterBackend,
     )
     filterset_class = PatientFilterSet
 
+    def get_queryset(self):
+        filter_query = self.request.GET.get('disease_status', None)
+        if (filter_query):
+            return PatientRegistration.objects.filter(
+                disease_status=DiseaseStatusEnum[filter_query].value).select_related(
+                "local_body", "district", "state", "facility",
```
Comment (vigneshhari) [inline care/facility/api/viewsets/patient.py:66]: Codacy Issue found: [Wrong hanging indentation (add 4 spaces).](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305809)

File: config/api_router.py [hunk 7]
```diff
@@ -24,7 +24,7 @@
 router.register("ambulance/create", AmbulanceCreateViewSet)
 router.register("ambulance", AmbulanceViewSet)
 
-router.register("patient", PatientViewSet)
+router.register("patient", PatientViewSet, basename='patient')
```
Comment (anroopak) [inline config/api_router.py:27]: Why is this required?

**Comment by anroopak** [comment]
> The code breaks all security validations in place right ?, correct me if i am wrong.

If you meant the usage of `request.GET` -> then yes.
Otherwise I don't see any problem
